.ci/check-performance-regression.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+set -e
+
+# max % difference tolerance
+MAX_DIFFERENCE=10
+
+# parse benchmark result at the given path
+parse_benchmark_result() {
+  BENCHMARK_PATH=$1
+  EXECUTION_TIME=$(awk '/Average Execution Time:/ {print $4}' "$BENCHMARK_PATH")
+
+  local RESULT_ARRAY=($EXECUTION_TIME)
+  echo "${RESULT_ARRAY[@]}"
+}
+
+# compare baseline and patch benchmarks
+compare_results() {
+  # Calculate percentage difference for execution time
+  EXECUTION_TIME_DIFFERENCE=$(echo "scale=4; \
+    ((${PATCH[0]} - ${BASELINE[0]}) / ${BASELINE[0]}) * 100" | bc)
+  echo "Execution Time Difference: $EXECUTION_TIME_DIFFERENCE%"
+
+  # Check if differences exceed the maximum allowed difference
+  if (( $(echo "$EXECUTION_TIME_DIFFERENCE > $MAX_DIFFERENCE" | bc -l) )); then
+    echo "Differences exceed the maximum allowed difference (${MAX_DIFFERENCE}%)!"
+    exit 1
+  else
+    echo "Differences are within the maximum allowed difference."
+    exit 0
+  fi
+}
+
+# parse baseline benchmark
+BASELINE=($(parse_benchmark_result "./.ci/baseline.benchmark"))
```

Comment:
We may want to read the baseline from the `master` branch later after this purposal is accepted.